src/routes/docs/products/ai/integrations/replicate/+page.markdoc

```diff
@@ -3,6 +3,369 @@ layout: article
+    });
+  }
+};
+```
+
+This function so far will serve out HTML File but will also throw an error if the `REPLICATE_API_KEY` environment variable is missing.
+{% /section %}
+
+{% section #step-6 step=6 title="Use the Replicate API" %}
+Now that we have our HTML file being served, we can begin to add the code that will talk to the replicate API.
+
+First, we're going to import the Replicate SDK at the top of our `main.js` file:
+
+```js
+import Replicate from "replicate";
+```
+
+Next after we serve the HTML we're going to add code to validate the body of the request, define our models and initialise the Replicate SDK:
+```js
+const models = {
+  'audio': 'meta/musicgen:b05b1dff1d8c6dc63d14b0cdb42135378dcb87f6373b0d3d341ede46e59e2b38',
+  'text': 'meta/llama-2-70b-chat',
+  'image': 'stability-ai/sdxl:39ed52f2a78e934b3ba6e2a89f5b1c712de7dfea535525255b1aa35c5565e08b'
+};
+
+try {
+  throwIfMissing(req.body, ['prompt', 'type']);
+} catch (err) {
+  return res.json({ ok: false, error: err.message }, 400);
+}
+
+if (req.body.type !== 'audio' && req.body.type !== 'text' && req.body.type !== 'image') {
+  return res.json({ ok: false, error: 'Invalid type' }, 400);
+}
+
+const replicate = new Replicate();
+```
+In this example we're going to be using meta's musicgen and llama2 70b models for music and text generation while using Stability AI's SDXL model for image generation. You can find more models on the [Replicate explore page](https://replicate.com/explore).
+
+Next we're going to add some per model configurations:
+
+```js
+let request = {
+  input: {
+    prompt: req.body.prompt,
+  }
+};
+
+// Allows you to tinker parameters for individual output types
+switch (req.body.type) {
+  case 'audio':
+    request.input = {
+      ...request.input,
+      length: 30,
+    }
+  break;
+  case 'text':
+    request.input = {
+      ...request.input,
+      max_new_tokens: 512,
+    }
+  break;
+  case 'image':
+    request.input = {
+      ...request.input,
+      width: 512,
+      height: 512,
+      negative_prompt: "deformed, noisy, blurry, distorted",
+    }
+  break;
+};
+```
+This allows us to individually configure each of the models we're using, feel free to play with this configuration to get the best results for your use case.
+
+Finally with our request built we can call the replicate API and generate a prediction:
+```js
+let response;
+
+try {
+  response = await replicate.run(models[req.body.type], request);
+} catch (err) {
+  error(err);
+
+  return res.json({ ok: false, error: 'Failed to run model' }, 500);
+}
+
+if (req.body.type === 'image') {
+  response = response[0]
+} else if (req.body.type === 'text') {
+  response = response.join('');
+}
+
+return res.json({ ok: true, response, type: req.body.type }, 200);
+```
+
+We've also added some error handling and response formatting to make it easier to work with the API.
+{% /section %}
+
+{% section #step-7 step=7 title="Making our page interactive" %}
```

Comment:
Like I mentioned in the perplexity docs, please combine steps 4,5,7. One step for all GET related stuff to serve a web page and 1 step for all the POST stuff to handle all the POST actions